CMakeLists.txt

```diff
@@ -171,7 +171,7 @@ function(avif_target_link_library target)
     # The transitive dependency is an export link library if it is a shared
     # library, or if it is a static library in a static build
     get_target_property(target_type ${target} TYPE)
-    if(target_type STREQUAL "SHARED_LIBRARY" OR NOT BUILD_SHARED_LIBS)
+    if(NOT BUILD_SHARED_LIBS)
```

Comment:
1) Also delete line 173. It sets the `target_type` variable, which is now unused.

2) Please review and update the comments at line 158 and line 171.

It seems that the comment at line 158 is not affected by the fix, although I don't really understand what "for export"in line 159 means. It seems to be in contrast to "for build" in line 158, but if that's the case, then line 159 should read "the <target> library to avif's public link library for install if a consumer". I.e., the two comment lines describe line 162:

```
    target_link_libraries(avif_obj PUBLIC $<BUILD_INTERFACE:${target}>)
```

respectively.

and line 175:

```
        target_link_libraries(avif PUBLIC $<INSTALL_INTERFACE:${install_target}>)
```

The comment at line 171 is affected by this change, so it should be updated. It probably just needs to say we need a transitive dependency only in a static build, or we can simply delete the comment. (Ideally the comment should explain why we do not need a transitive dependency in a shared build.)

Reply:
I believe the new comments I added are right.

Reply:
Done